deck-of-cards-kata/src/main/java/bnymellon/codekatas/deckofcards/custom/collections/HashBag.java

```diff
@@ -21,164 +21,195 @@
 import java.util.function.BiConsumer;
 import java.util.function.Consumer;
 
-public class HashBag<T> implements MutableBag<T> {
+public class HashBag<T> implements MutableBag<T>
+{
     private MutableMap<T, Integer> backingMap = MutableMap.empty();
     private int size = 0;
 
-    HashBag<T> withAll(T... elements) {
-        for (T element : elements) {
+    HashBag<T> withAll(T... elements)
+    {
+        for (T element : elements)
+        {
             this.backingMap.merge(element, 1, (existingValue, newValue) -> existingValue + 1);
             this.size++;
         }
         return this;
     }
 
     @Override
-    public int sizeDistinct() {
+    public int sizeDistinct()
+    {
         return this.backingMap.size();
     }
 
     @Override
-    public int size() {
+    public int size()
+    {
         return this.size;
     }
 
     @Override
-    public boolean isEmpty() {
+    public boolean isEmpty()
+    {
         return this.size == 0;
     }
 
     @Override
-    public boolean contains(Object o) {
+    public boolean contains(Object o)
+    {
         return this.backingMap.containsKey(o);
     }
 
     @Override
-    public Object[] toArray() {
+    public Object[] toArray()
+    {
         Object[] result = new Object[this.size()];
         this.forEachWithIndex((each, index) -> result[index] = each);
         return result;
     }
 
-    public void forEachWithIndex(BiConsumer<? super T, Integer> biConsumer) {
+    public void forEachWithIndex(BiConsumer<? super T, Integer> biConsumer)
+    {
         Counter index = new Counter();
         this.backingMap.forEach((key, count) ->
         {
-            for (int i = 0; i < count; i++) {
+            for (int i = 0; i < count; i++)
+            {
                 biConsumer.accept(key, index.getCount());
                 index.increment();
             }
         });
     }
 
     @Override
-    public <T1> T1[] toArray(T1[] array) {
+    public <T1> T1[] toArray(T1[] array)
+    {
         int size = this.size();
         T1[] result = array.length < size
                 ? (T1[]) Array.newInstance(array.getClass().getComponentType(), size)
                 : array;
 
         this.forEachWithIndex((each, index) -> result[index] = (T1) each);
-        if (result.length > size) {
+        if (result.length > size)
+        {
             result[size] = null;
         }
         return result;
     }
 
     @Override
-    public boolean add(T element) {
-        int sizeBefore = this.size;
+    public boolean add(T element)
+    {
         this.backingMap.merge(element, 1, (existingValue, newValue) -> existingValue + 1);
-        return sizeBefore != this.size;
+        this.size++;
+        return true;
     }
 
     @Override
-    public boolean remove(Object o) {
+    public boolean remove(Object o)
+    {
         return this.removeOccurrences((T) o, Integer.MAX_VALUE);
```

Comment:
this can be changed to `this.backingMap.remove(o)`

Reply:
This is a HashBag improvement and should be in a separate PR. I tried not changing anything existing unless it was necessary.